File: backend/utils/create_token.py
import jwt
import datetime
from config import config

SECRET_KEY = config["JWT_SECRET"]   
ALGORITHM = "HS256"

def create_token(email: str) -> str:
    """Generate JWT token for a given email"""
    payload = {
        "sub": email,
        "exp": datetime.datetime.utcnow() + datetime.timedelta(hours=12),  # expires in 12h
        "iat": datetime.datetime.utcnow()
    }
    token = jwt.encode(payload, SECRET_KEY, algorithm=ALGORITHM)
    return token

File: backend/src/utils/createToken.ts
import jwt from "jsonwebtoken"
import config from "../config"

export const createToken = (email: string) => {
    return jwt.sign({ email }, config["JWT_SECRET"] as string, { expiresIn: "7d" })
}

export const verifyToken = async (token: string) => {
    try {
        const decoded = jwt.verify(token, config["JWT_SECRET"] as string) as { email: string }
        return decoded.email
    } catch (err) {
        return null
    }
}